src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepositorySpec.scala

```diff
@@ -87,6 +87,86 @@ class FileRepositorySpec extends TestContainerUtils with ScalaFutures with Match
+      val folderIdTwo = UUID.fromString("2dfa0495-72a3-4e88-9c0e-b105d7802a4e")
+
+      utils.createConsignment(consignmentId, userId)
+
+      utils.createFile(folderIdOne, consignmentId, NodeType.directoryTypeIdentifier)
+      utils.createFile(UUID.fromString("d870fb86-0dd5-4025-98d3-11232690918b"), consignmentId, parentId = Some(folderIdOne))
+      utils.createFile(folderIdTwo, consignmentId, NodeType.directoryTypeIdentifier)
+      utils.createFile(UUID.fromString("317c7084-d3d4-435b-acd2-cfb317793843"), consignmentId,  parentId = Some(folderIdTwo))
+
+      val consignmentFiles = fileRepository.countFilesInConsignment(consignmentId, fileTypeIdentifier = None).futureValue
+
+      consignmentFiles shouldBe 4
+  }
+
+  "countFilesInConsignment" should "return the total number of folders in a consignment given a folder type filter" in withContainers {
+    case container: PostgreSQLContainer =>
+      val db = container.database
+      val utils = TestUtils(db)
+      val fileRepository = new FileRepository(db)
+      val consignmentId = UUID.fromString("2e31c0ce-25e6-4bd7-a8a7-dc8bbb9335ba")
+
+      utils.createConsignment(consignmentId, userId)
+
+      utils.createFile(UUID.fromString("4bde68aa-6212-45dc-9097-769b9f77dbd9"), consignmentId, NodeType.directoryTypeIdentifier)
+      utils.createFile(UUID.fromString("d870fb86-0dd5-4025-98d3-11232690918b"), consignmentId)
+      utils.createFile(UUID.fromString("2dfa0495-72a3-4e88-9c0e-b105d7802a4e"), consignmentId)
+      utils.createFile(UUID.fromString("1ad53749-aba4-4369-8fd6-2311111427cc"), consignmentId)
+
+      val consignmentFiles = fileRepository.countFilesInConsignment(consignmentId,
+        fileTypeIdentifier = Some(NodeType.directoryTypeIdentifier)).futureValue
+
+      consignmentFiles shouldBe 1
+  }
+
+  "countFilesInConsignment" should "return the total number of files in a consignment given a file type filter" in withContainers {
+    case container: PostgreSQLContainer =>
+      val db = container.database
+      val utils = TestUtils(db)
+      val fileRepository = new FileRepository(db)
+      val consignmentId = UUID.fromString("2e31c0ce-25e6-4bd7-a8a7-dc8bbb9335ba")
+
+      utils.createConsignment(consignmentId, userId)
+
+      utils.createFile(UUID.fromString("4bde68aa-6212-45dc-9097-769b9f77dbd9"), consignmentId, NodeType.directoryTypeIdentifier)
+      utils.createFile(UUID.fromString("d870fb86-0dd5-4025-98d3-11232690918b"), consignmentId)
+      utils.createFile(UUID.fromString("2dfa0495-72a3-4e88-9c0e-b105d7802a4e"), consignmentId)
+      utils.createFile(UUID.fromString("1ad53749-aba4-4369-8fd6-2311111427cc"), consignmentId)
+
+      val consignmentFiles = fileRepository.countFilesInConsignment(consignmentId,
+        fileTypeIdentifier = Some(NodeType.fileTypeIdentifier)).futureValue
+
+      consignmentFiles shouldBe 3
+  }
+
+  "countFilesInConsignment" should "return the total number of files in a consignment given a parentId filter" in withContainers {
```

Comment:
This test should be rename: `... "return the total number of files and folders with the same parent given a parentId filter"`